filterpy/kalman/kalman_filter.py

```diff
@@ -304,49 +304,57 @@ def predict(self, u=0):
         self._P = self._alpha_sq * dot3(self._F, self._P, self._F.T) + self._Q
 
 
-    def batch_filter(self, zs, Rs=None, update_first=False):
+    def batch_filter(self, zs, Rs=None, Fs=None, Hs=None, Qs=None, update_first=False):
         """ Batch processes a sequences of measurements.
-
         **Parameters**
-
         zs : list-like
             list of measurements at each time step `self.dt` Missing
             measurements must be represented by 'None'.
-
         Rs : list-like, optional
             optional list of values to use for the measurement error
             covariance; a value of None in any position will cause the filter
             to use `self.R` for that time step.
-
+        Fs : list-like, optional
+            optional list of values to use for the update matrix;
+            a value of None in any position will cause the filter
+            to use `self.R` for that time step.
+        Hs : list-like, optional
+            optional list of values to use for the measurement matrix;
+            a value of None in any position will cause the filter
+            to use `self.R` for that time step.
+        Qs : list-like, optional
+            optional list of values to use for the process error
+            covariance; a value of None in any position will cause the filter
+            to use `self.R` for that time step.
         update_first : bool, optional,
             controls whether the order of operations is update followed by
             predict, or predict followed by update. Default is predict->update.
-
         **Returns**
-
-
         means: np.array((n,dim_x,1))
             array of the state for each time step after the update. Each entry
             is an np.array. In other words `means[k,:]` is the state at step
             `k`.
-
         covariance: np.array((n,dim_x,dim_x))
             array of the covariances for each time step after the update.
             In other words `covariance[k,:,:]` is the covariance at step `k`.
-
         means_predictions: np.array((n,dim_x,1))
             array of the state for each time step after the predictions. Each
             entry is an np.array. In other words `means[k,:]` is the state at
             step `k`.
-
         covariance_predictions: np.array((n,dim_x,dim_x))
             array of the covariances for each time step after the prediction.
             In other words `covariance[k,:,:]` is the covariance at step `k`.
         """
 
         n = np.size(zs,0)
-        if Rs is None:
-            Rs = [None]*n
+        if np.all(Rs == None):
```

Comment:
These checks look a little strange to me, I think something like `if Rs is None` is the more canonical way to check for None-ness